ext/CUDASupportExt.jl

```diff
@@ -0,0 +1,26 @@
+module CUDASupportExt
+using CUDA 
+using Adapt
+using ShiftedArrays
+using Base # to allow displaying such arrays without causing the single indexing CUDA error
+
+Adapt.adapt_structure(to, x::CircShiftedArray{T, D, CT}) where {T,D,CT<:CuArray} = CircShiftedArray(adapt(to, parent(x)), shifts(x));
+function Base.Broadcast.BroadcastStyle(::Type{T})  where (T<: CircShiftedArray{<:Any,<:Any,<:CuArray})
+    CUDA.CuArrayStyle{ndims(T)}()
+end
+
+Adapt.adapt_structure(to, x::ShiftedArray{T, M, N, <:CuArray}) where {T,M,N} =
+# lets do this for the ShiftedArray type
+ShiftedArray(adapt(to, parent(x)), shifts(x); default=ShiftedArrays.default(x))
+function Base.Broadcast.BroadcastStyle(::Type{T})  where (T<: ShiftedArray{<:Any,<:Any,<:Any,<:CuArray})
+    CUDA.CuArrayStyle{ndims(T)}()
+end
+
+function Base.show(io::IO, mm::MIME"text/plain", cs::CircShiftedArray) 
+    CUDA.@allowscalar invoke(Base.show, Tuple{IO, typeof(mm), AbstractArray}, io, mm, cs) 
+end
+
+function Base.show(io::IO, mm::MIME"text/plain", cs::ShiftedArray) 
+    CUDA.@allowscalar invoke(Base.show, Tuple{IO, typeof(mm), AbstractArray}, io, mm, cs) 
```

Comment:
This part makes me a little uncomfortable (it's type-piracy in some sense), maybe it's best to remove it for the time being and accept that showing errors and we can see how to fix it later. How do other wrapping packages do it? Should we maybe just depend on GPUArraysCore and take `@allowscalar` from there?

Reply:
I agree. I will commend this out for the time being and test your other great suggestions,